### PR TITLE
Modified attr name to get max ports supported on switch

### PR DESCRIPTION
**Pull request by AshokDaparthi**
Discussed as part of PR #487 and decided to add new attribute in master.  

Issue: 
Previous attribute SAI_SWITCH_ATTR_PORT_NUMBER is not clear.
is this for active ports or max supported ports by NPU ?
Max supported ports are more relevant, since apps can query SAI_SWITCH_ATTR_PORT_LIST to get active list or adapter will create/delete ports as part of switch_init.
They will have active port count in DB.

### CODE REVIEW

File: inc/saiswitch.h [hunk 1]
```diff
@@ -237,12 +237,20 @@ typedef enum _sai_switch_attr_t
...
 
     /**
-     * @brief The number of ports on the switch
+     * @brief Number of active ports on the switch
      *
      * @type sai_uint32_t
      * @flags READ_ONLY
      */
-    SAI_SWITCH_ATTR_PORT_NUMBER = SAI_SWITCH_ATTR_START,
+    SAI_SWITCH_ATTR_NUMBER_OF_ACTIVE_PORTS = SAI_SWITCH_ATTR_START,
```
Comment (lguohan) [inline inc/saiswitch.h]: can you define "active port"? what is the definition of "active port"?

Comment (lguohan) [inline inc/saiswitch.h]: for backward compatibility, please follow commit https://github.com/opencomputeproject/SAI/commit/3cc57f24b0d5df1b0e41abc5450cbbe2592352bf

Comment (AshokDaparthi) [inline inc/saiswitch.h]: Done

**Comment by lguohan** [comment]
use the @ignore for backward compatibility

**Comment by lguohan** [comment]
should be ACTIVE NUMBER of ports for backward compatiblity

File: inc/saiswitch.h [hunk 2]
```diff
@@ -1379,6 +1387,13 @@ typedef enum _sai_switch_attr_t
 #define SAI_KEY_HW_PORT_PROFILE_ID_CONFIG_FILE    "SAI_HW_PORT_PROFILE_ID_CONFIG_FILE"
 
 /**
+ * @def SAI_SWITCH_ATTR_PORT_NUMBER
+ *
+ * For backward compatibility.
+ */
+#define SAI_SWITCH_ATTR_PORT_NUMBER     SAI_SWITCH_ATTR_MAX_NUMBER_OF_SUPPORTED_PORTS
```
Comment (kcudnik) [inline inc/saiswitch.h]: declare this as enum and use @ignore  tag

**Pull request merged**